src/types/timezone.jl

```diff
@@ -64,16 +64,18 @@ US/Pacific (UTC-8/UTC-7)
 """
 TimeZone(::AbstractString, ::Class)
 
-function TimeZone(str::AbstractString, mask::Class=Class(:DEFAULT))
+TimeZone(str::AbstractString, args...) = TimeZone(String(str), args...)
+
+function TimeZone(str::String, mask::Class=Class(:DEFAULT))
     # Note: If the class `mask` does not match the time zone we'll still load the
     # information into the cache to ensure the result is consistent.
-    tz, class = get!(_tz_cache(), str) do
-        tz_path = joinpath(TZData.COMPILED_DIR, split(str, "/")...)
+    tz, class = get!(_tz_cache(), String(str)) do
+        tz_path = joinpath(TZData.COMPILED_DIR, split(String(str), "/")...)
 
         if isfile(tz_path)
             open(deserialize, tz_path, "r")
-        elseif occursin(FIXED_TIME_ZONE_REGEX, str)
-            FixedTimeZone(str), Class(:FIXED)
+        elseif occursin(FIXED_TIME_ZONE_REGEX, String(str))
+            FixedTimeZone(String(str)), Class(:FIXED)
```

Comment:
Is this required?
It seems like it is just going to be converted to a InlineString anyway so why convert to a String first?

Reply:
removed